uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-calendar.js

```diff
@@ -506,15 +530,16 @@ define([
      */
     function setCalendarProps(contactID, calendar) {
       var leaveRequest,
-        indexedByMonth = vm.months.map(function () {
-          return [];
+        monthObject,
+        indexedByMonth = _.map(vm.months, function (month) {
+          return _.extend(month, {
+            data: []
```

Comment:
you should define the `data` property in `fetchMonthsFromPeriod`, otherwise you are initializing 2/3 of the object there, and the remaning 1/3 here, it's confusing
```js
while (startDate.isBefore(endDate)) {
  months.push({
    month: startDate.month(),
    year: startDate.year(),
    data: []
  });

  startDate.add(1, 'month');
}
```